tests/rptest/services/redpanda.py

```diff
@@ -1278,9 +1278,14 @@ def redpanda_pid(self, node):
             return None
 
     def pids(self, node):
-        """Return process ids associated with running processes on the given node."""
+        """
+        Return process ids for the following processes on the given node:
+        * 'redpanda' started in 'start_redpanda'
+        * 'node' started in 'start_wasm_engine' (only when coproc is enabled)
+        """
         try:
-            cmd = "ps ax | grep -i 'redpanda\|node' | grep -v grep | awk '{print $1}'"
+            proc_name_regex = "redpanda\|/bin/node"
+            cmd = f"ps ax | grep -i '{proc_name_regex}' | grep -v grep | awk '{{print $1}}'"
```

Comment:
nice find. this is strictly better than what we had.

do you think that we should switch this to only include the process name and not include the command line arguments for the process:

```
[nwatkins@fedora redpanda]$ ps -a -o pid= -o comm= 
   2277 Xorg
   2347 gnome-session-b
  44555 tmux: client
  84383 vim
  85703 ps
```

for example what we have now looks at all of the arguments to the process (e.g. `node`v in the fusermount program). using `/bin/node` is unlikely to give us a false positive in this case, but maybe not so for a plain `redpanda`?

```
[nwatkins@fedora redpanda]$ ps ax | grep node
    919 ?        I<     0:00 [xfs-inodegc/dm-]
   2865 ?        Ss     0:00 fusermount -o rw,nosuid,nodev,fsname=portal,auto_unmount,subtype=portal -- /run/user/1000/doc
```

Reply:
I think omitting the arguments is fine as a way of reducing false positives. Additionally, we could use the fully qualified paths for 'redpanda' and 'node' in the regex.

Reply:
> fully qualified paths

This would be nice, but I'm not sure how flexible it will be. I would imagine that in general the paths will be different in docker vs cdt environments, even if they happen to be the same now.

Reply:
im also happy keeping it the way it is now. just let me know

Reply:
We [generate the path](https://github.com/redpanda-data/redpanda/blob/dev/tests/rptest/services/redpanda.py#L1163) based on the env, so I think it would be fine.

Reply:
Will update on Monday, but if you want to merge it in the current state I also don't mind.

Reply:
sounds good. ill merge so we get better CI behavior